### PR TITLE
refactor(cli-config): Break up cli config.ts into focused modules (Fixes #1582)

### LINKED ISSUE #1582: Break up cli config.ts (1,923 lines)
## Break up cli config.ts (1,923 lines)

packages/cli/src/config/config.ts is a 1,923-line file that needs decomposition.

### Proposed Decomposition

1. **CliArgParser** -- Parses and validates CLI arguments (yargs integration)
2. **ConfigFileHandler** -- Reads CLI-specific config files, handles file discovery
3. **ConfigMerger** -- Merges CLI args with file config and core config
4. **ProfileBootstrap** -- Profile discovery, selection, and initialization for CLI
5. **config.ts** (facade) -- Thin coordinator

### Notes

- Must be decomposed in coordination with core config.ts (separate sub-issue)
- The boundary between core config and cli config should be clarified: core handles settings/provider config, cli handles arg parsing and UI config
- Consider how this maps to the proposed packages/settings package

### Acceptance Criteria

- [ ] No single file exceeds 800 lines
- [ ] No single function exceeds 80 lines
- [ ] All existing tests pass
- [ ] Test coverage does not decrease

### PR DESCRIPTION
**Pull request by acoliver**
## TLDR

Decomposes the 2,023-line monolith packages/cli/src/config/config.ts into 12 focused single-responsibility modules. The original loadCliConfig function (1,100+ lines) is now a 55-line thin orchestrator that delegates to typed, tested modules.

**No backward-compatibility re-exports** per parent issue #1568 -- all callers updated to import from canonical module locations.

### Architecture

The original config.ts combined argument parsing, environment loading, profile resolution, approval mode logic, provider/model precedence, MCP server configuration, tool governance, Config object construction, and post-config runtime side effects in a single file. This PR extracts each concern into its own module:

| Module | Lines | Responsibility |
|--------|-------|---------------|
| config.ts (orchestrator) | 390 | Thin orchestrator with loadCliConfig (55 lines) |
| cliArgParser.ts | 299 | CliArgs interface, parseArguments function |
| yargsOptions.ts | 402 | Declarative yargs option definitions |
| environmentLoader.ts | 119 | findEnvFile, loadEnvironment, isDebugMode, loadHierarchicalLlxprtMemory |
| toolGovernance.ts | 114 | Tool normalization, exclusion filtering, merging |
| mcpServerConfig.ts | 67 | MCP server merging and filtering |
| approvalModeResolver.ts | 83 | Approval mode precedence resolution |
| providerModelResolver.ts | 93 | Provider/model 4-level and 6-level precedence chains |
| profileResolution.ts | 315 | Profile loading, preparation, synthetic profiles |
| interactiveContext.ts | 237 | Interactive mode, directory resolution, file filtering |
| configBuilder.ts | 301 | Config object construction with sub-builders |
| postConfigRuntime.ts | 439 | Post-Config side effects (steps 10-17) |

### Key Design Decisions

1. **No re-exports** from config.ts -- each module is the canonical import source
2. **Trust boundary preserved** -- originalSettings (not profile-merged) used for folder trust checks
3. **Dual registerCliProviderInfrastructure call preserved** -- once in bootstrap (step 2), once conditionally after runtime context (step 11)
4. **Dynamic import semantics preserved** -- lazy loading of runtimeSettings.js maintained
5. **17-step ordering documented** in both plan and code comments

### Parity Tests

8 new parity test files (182 tests total) lock the behavior BEFORE extraction:
- Approval mode precedence
- Provider/model precedence
- MCP server filtering
- Tool governance
- Argument parsing
- Profile override precedence
- End-to-end ordering
- Folder trust security boundary

## Reviewer Test Plan

1. Pull the branch and run the full verification:
   - npm run typecheck -- clean
   - npx vitest run packages/cli/src/config/__tests__/ -- all 182 parity tests pass
   - npm run build -- clean
   - node scripts/start.js --profile-load synthetic "write me a haiku and nothing else" -- works

2. Verify file sizes: wc -l packages/cli/src/config/*.ts -- no file >800 lines

3. Check the orchestrator is thin: read packages/cli/src/config/config.ts -- loadCliConfig is 55 lines

4. Verify no re-exports: grep "export.*from" packages/cli/src/config/config.ts -- empty

## Testing Matrix

|          | mac | win | linux |
| -------- | --- | --- | --- |
| npm run  | OK  | ?   | ?   |
| npx      | ?   | ?   | ?   |
| Docker   | ?   | ?   | ?   |
| Podman   | ?   | -   | -   |
| Seatbelt | ?   | -   | -   |

## Linked issues / bugs

Fixes #1582

### CODE REVIEW

File: packages/cli/src/config/__tests__/parseArgumentsParity.test.ts [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+/**
+ * Task 1.5 – parseArguments behavioral parity tests
+ *
+ * Locks yargs behavior that could regress during parser extraction:
+ *   - Subcommand exit behavior (mcp, hooks, extensions, skills)
+ *   - Positional prompt/query mapping
+ *   - Conflicting flag handling
+ *   - Array coercion behavior
+ *   - Boolean default handling
+ */
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/parseArgumentsParity.test.ts:16]: This is a scope expansion. The `parseArgumentsParity` tests focus on verifying that the extracted `parseArguments` function produces identical results to the original inline parsing. Subcommand exit behavior is handled by `handleSubcommandExit()` which is a direct extraction - adding integration tests for mcp/hooks/extensions subcommands would be new test coverage beyond what this refactor requires.

File: packages/cli/src/config/environmentLoader.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  settings: Settings,
+  extensions: GeminiCLIExtension[],
+  folderTrust: boolean,
+  memoryImportFormat: 'flat' | 'tree' = 'tree',
+  fileFilteringOptions?: FileFilteringOptions,
+): Promise<{ memoryContent: string; fileCount: number; filePaths: string[] }> {
+  // FIX: Use real, canonical paths for a reliable comparison to handle symlinks.
+  const realCwd = fs.realpathSync(path.resolve(currentWorkingDirectory));
+  const realHome = fs.realpathSync(path.resolve(homedir()));
+  const isHomeDirectory = realCwd === realHome;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/environmentLoader.ts]: The original code in `config.ts` uses the same `fs.realpathSync` calls without defensive handling. This is a pure refactor that preserves existing behavior; adding defensive error handling would be a behavioral change out of scope.

File: packages/cli/src/config/configBuilder.ts [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+  readonly filePaths: readonly string[];
+  readonly policyEngineConfig: PolicyEngineConfig;
+  readonly question: string;
+  readonly screenReader: boolean;
+  readonly useRipgrepSetting: boolean | undefined;
+  readonly mcpEnabled: boolean;
+  readonly extensionsEnabled: boolean;
+  readonly adminSkillsEnabled: boolean;
+  readonly outputFormat: OutputFormat;
+  readonly allowedTools: readonly string[];
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/configBuilder.ts]: This is a faithful reproduction of the original behavior. The original `config.ts` on `main` also passes `blockedMcpServers: undefined` at the equivalent line (line 1468: `blockedMcpServers: undefined, // Extension-based blocking handled elsewhere`). This refactor intentionally preserves the existing behavior; any change to actually thread `blockedMcpServers` through would be a separate enhancement outside this PR's scope.

File: packages/cli/src/config/interactiveContext.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+  let resolvedLoadMemoryFromIncludeDirectories =
+    cliLoadMemoryPreference ?? settingsLoadMemoryPreference ?? false;
+
+  if (
+    !resolvedLoadMemoryFromIncludeDirectories &&
+    includeDirectoriesProvided &&
+    cliLoadMemoryPreference === undefined &&
+    settingsLoadMemoryPreference !== true
+  ) {
+    resolvedLoadMemoryFromIncludeDirectories = true;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/interactiveContext.ts]: This is a faithful reproduction of the original behavior. The original `config.ts` on `main` uses the identical condition `settingsLoadMemoryPreference !== true` (line 1103). This refactor preserves the existing logic exactly; changing the semantics of this condition would be a behavioral change outside this PR's scope.

File: packages/cli/src/config/configBuilder.ts [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+    enableHooksUI: getEnableHooksUI(settings),
+    hooks: eventHooks,
+    onReload: async () => {
+      const refreshedSettings = loadSettings(cwd);
+      return {
+        disabledSkills: refreshedSettings.merged.skills?.disabled,
+        adminSkillsEnabled:
+          refreshedSettings.merged.admin?.skills?.enabled ?? adminSkillsEnabled,
+      };
+    },
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/configBuilder.ts:115]: The original `loadCliConfig` in `config.ts` does not have error handling around `loadSettings` in the `onReload` callback either. Adding error handling here would be a behavioral change, not part of this pure refactor. Filed as a potential follow-up improvement but out of scope for this PR.

File: packages/cli/src/config/__tests__/folderTrustOriginalSettingsParity.test.ts [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+    );
+
+    // Trust check must have happened with original settings
+    expect(isWorkspaceTrusted).toHaveBeenCalled();
+    // Trust check received settings with folderTrust=false (original, not profile-merged)
+    expect(callsWithTrustedFalse.some((v) => v === true)).toBe(true);
+
+    // Because folder is not trusted, approval mode is forced to DEFAULT
+    expect(config.getApprovalMode()).toBe(ApprovalMode.DEFAULT);
+  });
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/folderTrustOriginalSettingsParity.test.ts:369]: This is a scope expansion. The test validates that the untrusted-folder code path uses the original settings object (not a profile-merged copy). Adding a profile-merge scenario would be a new test, not parity coverage for the refactor. The existing parity tests already cover the profile-merge path in other test files.

File: packages/cli/src/config/__tests__/approvalModeParity.test.ts [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+      getApprovalMode({ admin: { secureModeEnabled: true } }),
+    ).rejects.toThrow(/YOLO mode.*disabled/i);
+  });
+
+  it('secureModeEnabled=true allows AUTO_EDIT', async () => {
+    process.argv = ['node', 'script.js', '--approval-mode', 'auto_edit'];
+    expect(await getApprovalMode({ admin: { secureModeEnabled: true } })).toBe(
+      ApprovalMode.AUTO_EDIT,
+    );
+  });
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/approvalModeParity.test.ts:401]: This is a scope expansion. The parity tests verify the refactored code matches original behavior. Adding new combination scenarios (admin-disabled YOLO + untrusted folder) goes beyond verifying the refactor and would be new test coverage. The ordering of checks is preserved from the original code.

File: packages/cli/src/config/__tests__/toolGovernanceParity.test.ts [hunk 8]
```diff
@@ -0,0 +1,525 @@
...
+      '--allowed-tools',
+      'read_file',
+    ]);
+    const allowed = config.getEphemeralSetting('tools.allowed') as
+      | string[]
+      | undefined;
+    expect(allowed).toBeDefined();
+    const normalizedAllowed = allowed!.map((t) => t.trim().toLowerCase());
+    expect(normalizedAllowed).toContain('read_file');
+  });
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/toolGovernanceParity.test.ts:518]: Scope expansion. The tests verify the existing behavior is preserved through the refactor. Improving assertion specificity for allowlist merging would be new test design work beyond the scope of this refactor PR.

File: packages/cli/src/config/profileResolution.ts [hunk 9]
```diff
@@ -0,0 +1,317 @@
...
+      ({
+        effectiveSettings,
+        profileModel,
+        profileProvider,
+        profileModelParams,
+        profileBaseUrl,
+        loadedProfile,
+      } = result);
+    }
+  }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/profileResolution.ts:306]: The original code has the same structure. The caller already ensures inline profile takes precedence (inline profile is checked first in the calling code). Adding a redundant guard here would be a behavioral change, not part of this refactor.

File: packages/cli/src/config/__tests__/folderTrustOriginalSettingsParity.test.ts [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+
+  it('isWorkspaceTrusted is called exactly once for trust determination', async () => {
+    vi.mocked(isWorkspaceTrusted).mockReturnValue(true);
+    process.argv = ['node', 'script.js'];
+    const argv = await parseArguments({} as Settings);
+    await loadCliConfig({}, [], makeExtMgr(), 'test-session', argv);
+
+    // Called at least once for trust determination in loadCliConfig
+    expect(isWorkspaceTrusted).toHaveBeenCalled();
+  });
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/folderTrustOriginalSettingsParity.test.ts:377]: Good catch. Updated the assertion to `toHaveBeenCalledTimes(1)` to match the test name. Verified the test still passes.

File: packages/cli/src/config/__tests__/profileOverridePrecedenceParity.test.ts [hunk 11]
```diff
@@ -0,0 +1,526 @@
...
+    getRuntimeDiagnosticsSnapshot: vi.fn(() => ({})),
+    getActiveToolFormatState: vi.fn(() => ({})),
+    setActiveToolFormatOverride: vi.fn(),
+    getActiveProviderMetrics: vi.fn(() => undefined),
+    getSessionTokenUsage: vi.fn(() => undefined),
+    getLoadBalancerStats: vi.fn(() => undefined),
+    getLoadBalancerLastSelected: vi.fn(() => undefined),
+    getAllLoadBalancerStats: vi.fn(() => ({})),
+  };
+});
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/profileOverridePrecedenceParity.test.ts:306]: Trivial nitpick - the dual mocking is intentional since config.ts imports from both paths. Adding a comment is outside this fix commit scope.

File: packages/cli/src/config/__tests__/providerModelPrecedenceParity.test.ts [hunk 12]
```diff
@@ -0,0 +1,399 @@
...
+    // Clear all stubs then re-stub only what we need
+    vi.unstubAllEnvs();
+    vi.stubEnv('GEMINI_API_KEY', 'test-api-key');
+    // Remove provider/model env vars that may leak from CI environment
+    delete process.env.LLXPRT_DEFAULT_PROVIDER;
+    delete process.env.LLXPRT_DEFAULT_MODEL;
+    delete process.env.GEMINI_MODEL;
+    // Provide a fallback model so non-gemini providers don't fail with model.missing
+    vi.stubEnv('LLXPRT_DEFAULT_MODEL', 'mock-default-model');
+  });
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/providerModelPrecedenceParity.test.ts:302]: Fair point on the redundancy. The first stubEnv at line 280 was from the original code; the unstubAllEnvs + re-stub pattern was added to clear CI env leakage. Cleaning this up is a minor style improvement outside this fix scope.

File: packages/cli/src/config/__tests__/providerModelPrecedenceParity.test.ts [hunk 13]
```diff
@@ -0,0 +1,399 @@
...
+    const config = await runConfig({});
+    expect(config.getModel()).toBe('gemini-env-model');
+  });
+
+  it('level 5: LLXPRT_DEFAULT_MODEL beats GEMINI_MODEL', async () => {
+    vi.stubEnv('LLXPRT_DEFAULT_MODEL', 'preferred-model');
+    vi.stubEnv('GEMINI_MODEL', 'fallback-model');
+    const config = await runConfig({});
+    expect(config.getModel()).toBe('preferred-model');
+  });
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/providerModelPrecedenceParity.test.ts:385]: Good catch on the level numbering. This is purely a naming/documentation detail and does not affect test correctness. Outside the scope of this fix commit.

**Comment by acoliver** [comment]
@coderabbitai did it do the work or do we have a bunch of barrel export/imports and backward compatability shims?

File: packages/cli/src/config/config.ts [hunk 14]
```diff
@@ -6,800 +6,119 @@
...
-    nobrowser: result.nobrowser as boolean | undefined,
-    listSessions: result.listSessions as boolean | undefined,
-    deleteSession: result.deleteSession as string | undefined,
+import {
+  resolveProfileToLoad,
+  loadAndPrepareProfile,
+  type ProfileLoadResult,
+} from './profileResolution.js';
+import { resolveContextAndEnvironment } from './interactiveContext.js';
+import { resolveMemoryContent } from './environmentLoader.js';
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/config.ts]: Addressed. I restored environment loading by importing loadEnvironment from environmentLoader and calling loadEnvironment at the start of loadCliConfig before bootstrap and env dependent resolution. This restores nearest .llxprt/.env and .env support for LLXPRT_PROFILE, LLXPRT_DEFAULT_PROVIDER, LLXPRT_DEFAULT_MODEL, and related env backed settings.

File: packages/cli/src/config/mcpServerConfig.ts [hunk 15]
```diff
@@ -0,0 +1,112 @@
...
+    }
+    if (profileMergedSettings.excludeMCPServers) {
+      const excludedNames = new Set(
+        profileMergedSettings.excludeMCPServers.filter(Boolean),
+      );
+      if (excludedNames.size > 0) {
+        mcpServers = Object.fromEntries(
+          Object.entries(mcpServers).filter(([key]) => !excludedNames.has(key)),
+        );
+      }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/mcpServerConfig.ts:108]: Addressed. In mcpServerConfig I now append servers removed by profileMergedSettings.excludeMCPServers to blockedMcpServers before filtering mcpServers. This keeps excluded entries visible in getBlockedMcpServers and downstream MCP reporting instead of silently dropping them.

File: packages/cli/src/config/intermediateConfig.ts [hunk 16]
```diff
@@ -0,0 +1,113 @@
...
+  const allowedTools = argv.allowedTools || settings.allowedTools || [];
+  const allowedToolsSet = new Set(allowedTools);
+
+  let profileSettingsWithTools = profileMergedSettings;
+  if (allowedTools.length > 0) {
+    profileSettingsWithTools = {
+      ...profileMergedSettings,
+      tools: { ...profileMergedSettings.tools, allowed: allowedTools },
+    };
+  }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/intermediateConfig.ts]: Addressed. I changed allowed tools precedence to use argv.allowedTools first, then profileMergedSettings.allowedTools, then settings.allowedTools, and normalized via buildNormalizedToolSet before resolveNonInteractiveExcludes. This ensures profile merged allowlists are honored during non interactive exclusion resolution.

File: packages/lsp/test/lsp-client.test.ts [hunk 17]
```diff
@@ -308,7 +308,8 @@ describe('LspClient unit TDD edge cases and internal behaviors', () => {
     await client.waitForDiagnostics('/workspace/src/deadline-10.ts', 3000);
     const elapsed = Date.now() - start;
 
-    expect(elapsed).toBeLessThanOrEqual(3000);
+    // Allow small CI timer jitter (typically 1-5ms on loaded runners)
+    expect(elapsed).toBeLessThanOrEqual(3050);
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/lsp/test/lsp-client.test.ts:312]: No code change needed for this thread. The widened threshold and existing comment are already acceptable, and this nit is optional wording only.

File: packages/cli/src/config/toolGovernance.ts [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+  allowedTools: string[],
+  allowedToolsSet: Set<string>,
+): (tool: string) => boolean {
+  return (tool: string): boolean => {
+    if (tool === ShellTool.Name) {
+      // If any of the allowed tools is ShellTool (even with subcommands), don't exclude it.
+      return !allowedTools.some((allowed) =>
+        SHELL_TOOL_NAMES.some((shellName) => allowed.startsWith(shellName)),
+      );
+    }
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/toolGovernance.ts:109]: Addressed. I replaced the raw startsWith shell allow check with normalized set membership. createToolExclusionFilter now relies on canonical normalized tool names, including shell aliases/subcommand forms, which prevents false positives like run_shell_commander and correctly honors canonical shell identifiers.

File: packages/cli/src/config/profileRuntimeApplication.ts [hunk 19]
```diff
@@ -0,0 +1,188 @@
...
+    ({
+      appliedResult,
+      resolvedProviderAfterProfile,
+      resolvedModelAfterProfile,
+      resolvedBaseUrlAfterProfile,
+    } = collectSnapshotResult(snapshotResult, mutableWarnings));
+    logger.debug(
+      () =>
+        `[bootstrap] Applied CLI auth -> provider=${resolvedProviderAfterProfile}, model=${resolvedModelAfterProfile}, baseUrl=${resolvedBaseUrlAfterProfile ?? 'default'}`,
+    );
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/profileRuntimeApplication.ts:156]: Addressed. In the synthetic CLI auth profile branch I now propagate resolvedProviderAfterProfile into resolvedFinalProvider after collectSnapshotResult. That ensures activateProviderAndProfile and switchActiveProvider use the provider actually resolved by applyProfileSnapshot rather than a stale pre snapshot value.

**Comment by acoliver** [comment]
Progress update on active CodeRabbit items: implemented fixes for all currently open threads in the working tree and completed full verification (test, lint, typecheck, format, build, smoke). I still need to post per-thread replies and resolve threads, but GitHub is returning a review-comment API constraint ("user_id can only have one pending review per pull request") when attempting inline thread replies. I will finalize by posting thread-level replies/resolutions after this pending-review constraint is cleared.

**Review by acoliver — COMMENTED** [review]
Submitting existing draft review comments so thread replies are published.

File: packages/cli/src/config/__tests__/approvalModeParity.test.ts [hunk 20]
```diff
@@ -0,0 +1,446 @@
...
+describe('approvalModeParity: approval mode resolution', () => {
+  const originalArgv = process.argv;
+
+  beforeEach(() => {
+    vi.resetAllMocks();
+    vi.mocked(os.homedir).mockReturnValue(
+      path.resolve(path.sep, 'mock', 'home', 'user'),
+    );
+    vi.stubEnv('GEMINI_API_KEY', 'test-api-key');
+    process.argv = ['node', 'script.js'];
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/approvalModeParity.test.ts:304]: Addressed in this branch. I added explicit setup-time env scrubbing for LLXPRT_PROFILE, LLXPRT_DEFAULT_PROVIDER, LLXPRT_DEFAULT_MODEL, and GEMINI_MODEL in approvalModeParity before invoking parseArguments/loadCliConfig. This makes the approval-mode parity path deterministic regardless of inherited runner env state.

File: packages/cli/src/config/__tests__/profileOverridePrecedenceParity.test.ts [hunk 21]
```diff
@@ -0,0 +1,561 @@
...
+    vi.mocked(os.homedir).mockReturnValue(
+      path.resolve(path.sep, 'mock', 'home', 'user'),
+    );
+    vi.stubEnv('GEMINI_API_KEY', 'test-api-key');
+    // Remove provider/model env vars that may leak from CI environment
+    delete process.env.LLXPRT_DEFAULT_PROVIDER;
+    delete process.env.LLXPRT_DEFAULT_MODEL;
+    delete process.env.GEMINI_MODEL;
+    // Provide a fallback model so non-gemini providers don't fail with model.missing
+    vi.stubEnv('LLXPRT_DEFAULT_MODEL', 'mock-default-model');
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/profileOverridePrecedenceParity.test.ts:393]: Addressed. I now delete LLXPRT_PROFILE in the shared setup blocks in this file (including the additional setup locations called out in the thread) so tests that assume no env profile are not affected by inherited environment state.

File: packages/cli/src/config/__tests__/providerModelPrecedenceParity.test.ts [hunk 22]
```diff
@@ -0,0 +1,408 @@
...
+    runtimeSettingsState.oauthManager = null;
+    // Clear all stubs then re-stub only what we need
+    vi.unstubAllEnvs();
+    vi.stubEnv('GEMINI_API_KEY', 'test-api-key');
+    // Remove provider/model env vars that may leak from CI environment
+    delete process.env.LLXPRT_DEFAULT_PROVIDER;
+    delete process.env.LLXPRT_DEFAULT_MODEL;
+    delete process.env.GEMINI_MODEL;
+    // Provide a fallback model so non-gemini providers don't fail with model.missing
+    vi.stubEnv('LLXPRT_DEFAULT_MODEL', 'mock-default-model');
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/providerModelPrecedenceParity.test.ts]: Addressed. I added LLXPRT_PROFILE scrubbing in both setup blocks for this suite, alongside the existing provider/model env cleanup, so provider/model precedence assertions are isolated from inherited profile environment configuration.

File: packages/cli/src/config/__tests__/toolGovernanceParity.test.ts [hunk 23]
```diff
@@ -0,0 +1,556 @@
...
+  const originalArgv = process.argv;
+  const originalIsTTY = process.stdin.isTTY;
+
+  beforeEach(() => {
+    vi.resetAllMocks();
+    vi.mocked(os.homedir).mockReturnValue(
+      path.resolve(path.sep, 'mock', 'home', 'user'),
+    );
+    vi.stubEnv('GEMINI_API_KEY', 'test-api-key');
+    vi.mocked(isWorkspaceTrusted).mockReturnValue(true);
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/__tests__/toolGovernanceParity.test.ts:289]: Addressed. I added setup-time scrubbing for LLXPRT_PROFILE, LLXPRT_DEFAULT_PROVIDER, LLXPRT_DEFAULT_MODEL, and GEMINI_MODEL in each describe block highlighted in this suite so governance parity assertions run against only per-test inputs and not inherited environment state.

File: packages/cli/src/config/profileRuntimeApplication.ts [hunk 24]
```diff
@@ -0,0 +1,194 @@
...
+      syntheticProfile.ephemeralSettings['auth-key'] =
+        bootstrapArgs.keyOverride;
+    }
+    if (bootstrapArgs.keyfileOverride) {
+      syntheticProfile.ephemeralSettings['auth-keyfile'] =
+        bootstrapArgs.keyfileOverride;
+    }
+    if (bootstrapArgs.baseurlOverride) {
+      syntheticProfile.ephemeralSettings['base-url'] =
+        bootstrapArgs.baseurlOverride;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/profileRuntimeApplication.ts:134]: Addressed. In the synthetic profile path I now include keyNameOverride in the activation condition and propagate it to syntheticProfile.ephemeralSettings as auth-key-name. key-name-only and mixed auth-override cases now carry through the synthetic profile application path correctly.

File: packages/cli/src/config/intermediateConfig.ts [hunk 25]
```diff
@@ -0,0 +1,120 @@
...
+    argv.screenReader !== undefined
+      ? argv.screenReader
+      : (profileMergedSettings.accessibility?.screenReader ?? false);
+
+  const allowedTools =
+    argv.allowedTools ||
+    profileMergedSettings.allowedTools ||
+    settings.allowedTools ||
+    [];
+  const allowedToolsSet = buildNormalizedToolSet(allowedTools);
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/intermediateConfig.ts:59]: Addressed. resolveIntermediateConfig now considers both nested and legacy allowlist shapes in precedence order: argv.allowedTools, profileMergedSettings.tools.allowed, profileMergedSettings.allowedTools, settings.tools.allowed, settings.allowedTools. This keeps non-interactive exclusion resolution aligned with policy config when settings use nested tools.allowed.

File: packages/cli/src/config/postConfigRuntime.ts [hunk 26]
```diff
@@ -0,0 +1,461 @@
...
+      }
+    }
+  }
+
+  const cliSetResult = applyCliSetArguments(config, argv.set);
+
+  if (Object.keys(cliSetResult.modelParams).length > 0) {
+    (
+      config as Config & { _cliModelParams?: Record<string, unknown> }
+    )._cliModelParams = cliSetResult.modelParams;
```
[the remark this answers is not in the document]
Comment (acoliver) [reply packages/cli/src/config/postConfigRuntime.ts]: Addressed. applyEphemeralSettings now protects governance-managed keys from /set in non-interactive mode by filtering tools.allowed, tools.disabled, and disabled-tools before applyCliSetArguments, then re-applying applyToolPolicies when those keys were attempted. Interactive mode intentionally keeps /set governance control, preserving existing interactive behavior and associated tests.
